### PR TITLE
Custom Mqtt Test Module: Restore E2E tests

### PR DESCRIPTION
**Pull request by and-rewsmith**
I accidentally broke end to end tests when adding a new mode to the custom mqtt test module. I took this as an opportunity to address some of Roberts comments on my cherry pick I didn't address before. Also fixed the README.

### CODE REVIEW

File: test/Microsoft.Azure.Devices.Edge.Test/GenericMqtt.cs
Comment (and-rewsmith) [inline test/Microsoft.Azure.Devices.Edge.Test/GenericMqtt.cs:67]: This is the main fix

**Comment by and-rewsmith** [comment]
Tested the E2E test manually using a custom built generic-mqtt-tester image with the changes in this PR

**Review by vadim-kovalyov — APPROVED** [review]

**Comment by and-rewsmith** [comment]
Running e2e before merge anyway just to be super safe

**Comment by and-rewsmith** [comment]
E2E running here, waiting for pass before merge:
https://msazure.visualstudio.com/One/_build?definitionId=87020&_a=summary

**Comment by and-rewsmith** [comment]
@vadim-kovalyov Bridge config update is failing in my e2e test run:
https://msazure.visualstudio.com/One/_build/results?buildId=41942071&view=ms.vss-test-web.build-test-results-tab&runId=767393438&resultId=100035&paneView=asg-cloudtest.asg-cloudtest-attachments.asg-cloudtest.asg-cloudtest-attachments&path=e2e-fv-az207-556-210428-223709.524-bridgeconfigupdatetest-edgeHub.log

But it was passing in my earlier change:
https://github.com/Azure/iotedge/pull/4875#issuecomment-825073757

Any idea what is wrong?

**Pull request merged**
